frame/support/procedural/src/construct_runtime/expand/config.rs

```diff
@@ -97,6 +99,12 @@ pub fn expand_outer_config(
 				Ok(())
 			}
 		}
+
+		impl<T> #scrate::traits::GenesisBuild<T> for RuntimeGenesisConfig {
```

Comment:
I feel it is bit misuse of the trait, `GenesisBuild` was meant for pallet to declare their genesis build, thus the signature is specific `<T, I>`, and the specific doc for pallet usage.

I know it is good to abstract stuff into traits and reuse them but I feel the declaration of pallet genesis build is quite different from runtime so in the long term I'm not sure it makes sense to use a single trait.
Also this PR https://github.com/paritytech/substrate/pull/14330 seems to confirm my point.

Maybe we can extend `BuildStorage` with a `build_inside_ext` method (which would be the only method available in no-std) or create a new trait `RuntimeGenesisBuild` with a method `build`?

Reply:
I did not want to add new trait. But argument with `<T,I>` designed specifically for pallet is fair.
I've added new trait: https://github.com/paritytech/substrate/pull/14306/commits/3c131b618138ced29c01ab8d15d8c6410c9e128b

Reply:
AFAIK if it is used both by client and runtime then it should be in primitives probably , otherwise it can stay in frame_support

Reply:
This is intended to be used in runtime only.